test/cypress/integration/agreements/agreement.cy.js

```diff
@@ -42,7 +42,49 @@ describe('Agreement detail page', () => {
     cy.setEncryptedCookies(userExternalId)
   })
 
-  it('should display agreement detail page and allow cancelling agreement', () => {
+  it('should display agreement detail page', () => {
+    cy.task('setupStubs', [
+      ...userAndGatewayAccountStubs(),
+      agreementStubs.getLedgerAgreementSuccess({
+        service_id: serviceExternalId,
+        live: false,
+        gatewayAccountId,
+        external_id: 'a-valid-agreement-id'
+      }),
+      transactionsStub
+    ])
+
+    cy.visit('/test/service/service-id/account/gateway-account-id/agreements/a-valid-agreement-id')
+
+    cy.get('h1').contains('Agreement detail')
+
+    cy.get('.govuk-heading-l').should('have.text', 'Agreement detail')
+
+    cy.get('.govuk-summary-list__value').contains('Test User')
```

Comment:
It's not clear which `.govuk-summary-list__value` this is checking, which might make it difficult to debug if it starts failiing.

To be more deliberate about which elements we are checking the contents of, we could add a `data-cy` attribute on the elements in the Nunjucks and use that in the test to get the element we're checking.

See: https://pay-team-manual.cloudapps.digital/manual/how-to/write-cypress-tests.html#select-elements-with-the-data-cy-attribute

Also, are we checking all the attributes on the agreement are displayed as expected? Id, reference, status etc.

Reply:
👍